src/wallet/rpcwallet.cpp

```diff
@@ -2463,6 +2473,123 @@ UniValue fundrawtransaction(const UniValue& params, bool fHelp)
     return result;
 }
 
+UniValue bumpfee(const UniValue& params, bool fHelp)
+{
+    if (!EnsureWalletIsAvailable(fHelp))
+        return NullUniValue;
+
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+                            "bumpfee \"txid\"\n"
+                            "\nBumps the fee of a opt-in-RBF transaction.\n"
+                            "\nArguments:\n"
+                            "1. \"transactionid\"  (string, required) The transaction id\n"
+                            "\nResult:\n"
+                            "{\n"
+                            "  \"txid\":    \"value\", (string)  The id of the new transaction\n"
+                            "  \"oldfee\":    n,         (numeric) Fee of the replaced transaction\n"
+                            "  \"fee\":       n,         (numeric) Fee the resulting transaction pays\n"
+                            "}\n"
+                            "\nExamples:\n"
+                            "\nBump the fee, get the new transactions txid\n"
+                            + HelpExampleCli("bumpfee", "<txid>")
+                            );
+
+    RPCTypeCheck(params, boost::assign::list_of(UniValue::VSTR));
+    uint256 hash;
+    hash.SetHex(params[0].get_str());
+
+    // get the transaction
+    if (!pwalletMain->mapWallet.count(hash))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid or non-wallet transaction id");
+    const CWalletTx& wtx = pwalletMain->mapWallet[hash];
+
+    // ownly bump the fee if the transation signals opt-in-RBF after BIP125
+    if (!SignalsOptInRBF(wtx))
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Transaction can't be replaced");
+
+    // get the old fee to allow oldfee+newfee
+    CAmount nDebit = wtx.GetDebit(ISMINE_SPENDABLE);
+    CAmount nOldFee = -(wtx.IsFromMe(ISMINE_SPENDABLE) ? wtx.GetValueOut() - nDebit : 0);
+    CFeeRate oldFeeRate(nOldFee, (int)::GetSerializeSize((CTransaction)wtx, SER_NETWORK, PROTOCOL_VERSION));
+    CAmount nFee = 0;
+
+    CMutableTransaction tx(wtx);
+    // remove scriptSigs, the signatures are invalid after mutating the transaction
+    for (std::vector<CTxIn>::iterator it(tx.vin.begin()); it != tx.vin.end(); ++it) {
+        (*it).scriptSig = CScript();
+    }
+
+    // remove "old" change outputs
+    for (std::vector<CTxOut>::iterator it(tx.vout.begin()); it != tx.vout.end();)
+    {
+        // only remove IsMine outputs that are not in the address book (only change)
+        CTxDestination address;
+        if (pwalletMain->IsMine(*it) == ISMINE_SPENDABLE && (!ExtractDestination((*it).scriptPubKey, address) || !pwalletMain->mapAddressBook.count(address))) {
+            it = tx.vout.erase(it);
+        }
+        else {
+            ++it;
```

Comment:
I think this increment needs to happen always, even if the condition above is true.